src/main/java/org/jenkinsci/deprecatedusage/Downloader.java

```diff
@@ -48,7 +51,14 @@ public Downloader(ExecutorService executor, int maxConcurrentDownloads) {
                 }
                 download(file).handle((success, failure) -> {
                     if (failure != null) {
-                        failure.printStackTrace();
+                        // do not throw away the message!
+                        StringWriter sw = new StringWriter();
+                        PrintWriter pw = new PrintWriter(sw);
+                        pw.println("failure synching " + file.getName());
+                        pw.println(failure.getMessage());
+                        failure.printStackTrace(pw);
+                        pw.flush();
+                        System.err.println(sw.toString());
```

Comment:
jump through some hoops to try and print as atomically as we can to prevent the message being interleaved.